File: javascript/rotational-cipher/rotational-cipher.js
//
// This is only a SKELETON file for the 'Rotational Cipher' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const rotate = (text, rot) => {
  rot %= 26
  if (rot === 0) {
    return text
  }

  let cipher = ''
  for (const char of text.split('')) {
    let ordinance = char.charCodeAt()
    let div = null
    if (65 <= ordinance && ordinance <= 90) {
      div = 65
    }

    if (97 <= ordinance && ordinance <= 122) {
      div = 97
    }

    if (div === null) {
      cipher += char
      continue
    }
    
    ordinance -= div
    ordinance = (ordinance + rot) % 26
    ordinance += div
    cipher += String.fromCharCode(ordinance)
  }

  return cipher
}

File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    cipher = ''
    key %= 26 # ensure it's normalized over the alphabet

    for char in text:
        res = None
        ordinance = ord(char)
        if 65 <= ordinance <= 90:
            res = 65
        if 97 <= ordinance <= 122:
            res = 97

        if res is None:
            cipher += char
            continue

        ordinance -= res
        ordinance = (ordinance + key) % 26
        ordinance += res
        cipher += chr(ordinance)

    return cipher
        